tests/Hosting.Services.Web.UnitTests/Middlewares/UserIdentity/EmailBasedUserIdentityProviderTests.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.UnitTests
+{
+	[TestClass]
+	public class EmailBasedUserIdentityProviderTests
+	{
+		[TestMethod]
+		public async Task TryWriteBytes_ReturnFalseIfNoEmail()
+		{
+			IUserIdentityProvider provider = new EmailBasedUserIdentityProvider();
+			(HttpContext context, _) = HttpContextHelper.CreateHttpContext();
+			Memory<byte> memory = new byte[provider.MaxBytesInIdentity];
+			(bool result, int bytesWritten) = await provider.TryWriteBytesAsync(context, memory).ConfigureAwait(false);
+			Assert.IsFalse(result);
+			Assert.AreEqual(-1, bytesWritten);
+		}
+
+		[TestMethod]
+		public async Task TryWriteBytes_ChangedBaseOnEmail()
+		{
+			IUserIdentityProvider provider = new EmailBasedUserIdentityProvider();
+
+			HttpContext context1 = HttpContextHelper.GetContextWithEmail("Abc123@outlook.com");
+			HttpContext context2 = HttpContextHelper.GetContextWithEmail("Abc456@gmail.com");
+
+			byte[] hash1 = await GetIdentityAsync(provider, context1).ConfigureAwait(false);
```

Comment:
It's really overkill here, but FYI you can slightly improve the performance here using `Task.WhenAll` to await for the two results. That lets them execute in any order. But this is only a test so it doesn't matter much, and there are only two elements.

Reply:
I wouldn't worry about this comment. It was just a tip that might be helpful in the future, but it's of no great benefit here and these are just tests. So feel free just to close this off.